Gems/Atom/RHI/DX12/Code/Source/Platform/Windows/RHI/Device_Windows.cpp

```diff
@@ -222,18 +222,312 @@ namespace AZ
+                return "D3D12_AUTO_BREADCRUMB_OP_DECODEFRAME2";
+            case D3D12_AUTO_BREADCRUMB_OP_PROCESSFRAMES1:
+                return "D3D12_AUTO_BREADCRUMB_OP_PROCESSFRAMES1";
+            case D3D12_AUTO_BREADCRUMB_OP_BUILDRAYTRACINGACCELERATIONSTRUCTURE:
+                return "D3D12_AUTO_BREADCRUMB_OP_BUILDRAYTRACINGACCELERATIONSTRUCTURE";
+            case D3D12_AUTO_BREADCRUMB_OP_EMITRAYTRACINGACCELERATIONSTRUCTUREPOSTBUILDINFO:
+                return "D3D12_AUTO_BREADCRUMB_OP_EMITRAYTRACINGACCELERATIONSTRUCTUREPOSTBUILDINFO";
+            case D3D12_AUTO_BREADCRUMB_OP_COPYRAYTRACINGACCELERATIONSTRUCTURE:
+                return "D3D12_AUTO_BREADCRUMB_OP_COPYRAYTRACINGACCELERATIONSTRUCTURE";
+            case D3D12_AUTO_BREADCRUMB_OP_DISPATCHRAYS:
+                return "D3D12_AUTO_BREADCRUMB_OP_DISPATCHRAYS";
+            case D3D12_AUTO_BREADCRUMB_OP_INITIALIZEMETACOMMAND:
+                return "D3D12_AUTO_BREADCRUMB_OP_INITIALIZEMETACOMMAND";
+            case D3D12_AUTO_BREADCRUMB_OP_EXECUTEMETACOMMAND:
+                return "D3D12_AUTO_BREADCRUMB_OP_EXECUTEMETACOMMAND";
+            case D3D12_AUTO_BREADCRUMB_OP_ESTIMATEMOTION:
+                return "D3D12_AUTO_BREADCRUMB_OP_ESTIMATEMOTION";
+            case D3D12_AUTO_BREADCRUMB_OP_RESOLVEMOTIONVECTORHEAP:
+                return "D3D12_AUTO_BREADCRUMB_OP_RESOLVEMOTIONVECTORHEAP";
+            case D3D12_AUTO_BREADCRUMB_OP_SETPIPELINESTATE1:
+                return "D3D12_AUTO_BREADCRUMB_OP_SETPIPELINESTATE1";
+            case D3D12_AUTO_BREADCRUMB_OP_INITIALIZEEXTENSIONCOMMAND:
+                return "D3D12_AUTO_BREADCRUMB_OP_INITIALIZEEXTENSIONCOMMAND";
+            case D3D12_AUTO_BREADCRUMB_OP_EXECUTEEXTENSIONCOMMAND:
+                return "D3D12_AUTO_BREADCRUMB_OP_EXECUTEEXTENSIONCOMMAND";
+            case D3D12_AUTO_BREADCRUMB_OP_DISPATCHMESH:
+                return "D3D12_AUTO_BREADCRUMB_OP_DISPATCHMESH";
+            }
+            return "unkown op";
+        }
+
+        bool Device::AssertSuccess(HRESULT hr)
+        {
+            if (hr == DXGI_ERROR_DEVICE_REMOVED)
+            {
+                OnDeviceRemoved();
+            }
+
+            bool success = SUCCEEDED(hr);
+            AZ_Assert(success, "HRESULT not a success %x", hr);
+            return success;
+        }
+
+        void Device::OnDeviceRemoved()
+        {
+            AZStd::lock_guard<AZStd::mutex> lock(m_onDeviceRemovedMutex);
+
+            // It's possible this function is called many times at same time from different threads.
+            // We only want it be executed once.
+            if (m_onDeviceRemoved)
+            {
+                return;
+            }
+            m_onDeviceRemoved = true;
+
+            ID3D12Device* removedDevice = m_dx12Device.get();
+            HRESULT removedReason = removedDevice->GetDeviceRemovedReason();
+            
+            AZ_TracePrintf("Device", "Device was removed because of the following reason:\n");
```

Comment:
Suggest building a string here and sending the entire string to an `AZ_Crash` or `AZ_Fatal` log instead to make sure we have this data even if `AZ_ENABLE_TRACING` is off.